### PR TITLE
vim-patch:9.0.1620: Nix files are not recognized from the hashbang line

### PR DESCRIPTION
**Pull request by clason**
Problem:    Nix files are not recognized from the hashbang line.
Solution:   Add a hashbang check. (issue vim/vim#12507)

https://github.com/vim/vim/commit/19548c6a742d954ecd0b50b0680c37cc6ced7473

Co-authored-by: Bram Moolenaar <Bram@vim.org>

### CODE REVIEW

**Review by smjonas — APPROVED** [review]

**Comment by clason** [comment]
@smjonas Something is not working here -- maybe the fallback?

File: runtime/lua/vim/filetype/detect.lua [hunk 1]
```diff
@@ -1459,6 +1459,7 @@ local patterns_hashbang = {
   ['gforth\\>'] = { 'forth', { vim_regex = true } },
   ['icon\\>'] = { 'icon', { vim_regex = true } },
   guile = 'scheme',
+  ['nix-shell'] = 'nix',
```
Comment (seandewar) [inline runtime/lua/vim/filetype/detect.lua:1462]:
```suggestion
  ['nix%-shell'] = 'nix',
```

I think this also needs that `s:script_checks` change?

Comment (clason) [reply runtime/lua/vim/filetype/detect.lua]: keep forgetting that these are patterns...

locally this seems to suffice!

Comment (seandewar) [reply runtime/lua/vim/filetype/detect.lua]: Btw this is the change that's missing: https://github.com/vim/vim/commit/19548c6a742d954ecd0b50b0680c37cc6ced7473#diff-b8c15b633705ec2abf421ba8d16e5691fea9d4ae34318ac1d80f5f4b8fc4e358R855

Comment (clason) [reply runtime/lua/vim/filetype/detect.lua]: Wait, I didn't see that before. Added!

**Review by seandewar — APPROVED** [review]

**Pull request merged**